src/SonarScanner.MSBuild.Shim/PathHelper.cs

```diff
@@ -30,23 +30,19 @@ public static class PathHelper
     {
         public static string WithTrailingDirectorySeparator(this DirectoryInfo directory)
         {
-            if (directory == null)
-            {
-                throw new ArgumentNullException(nameof(directory));
-            }
-
-            if (directory.FullName.EndsWith(Path.DirectorySeparatorChar.ToString()) ||
-                directory.FullName.EndsWith(Path.AltDirectorySeparatorChar.ToString()))
+            _ = directory ?? throw new ArgumentNullException(nameof(directory));
```

Comment:
this cleanup can be on the other PR (or a different PR) , to separate cleanup from change of logic

Reply:
It's clean as you code. The changes aren't very big.
I've extracted the refactoring PR as that was a big change (moved methods, hard to review)